pkg/ipamd/datastore/data_store.go

```diff
@@ -349,59 +372,36 @@ func (ds *DataStore) ReadBackingStore() error {
 	ds.lock.Lock()
 	defer ds.lock.Unlock()
 
-	eniIPs := make(ENIPool)
-	eniPrefixes := make(ENIPool)
-
+	eniCidrs := make(ENIPool)
 	for _, eni := range ds.eniPool {
-		for _, addr := range eni.IPv4Addresses {
-			eniIPs[addr.Address] = eni
-		}
-		for _, prefix := range eni.IPv4Prefixes {
-			eniPrefixes[prefix.Prefix.IP.String()] = eni
+		//This will have /32 or /28 cidr to ENI mapping
+		for cidr, _ := range eni.AvailableIPv4Cidrs {
+			eniCidrs[cidr] = eni
 		}
 	}
 
 	for _, allocation := range data.Allocations {
-		eni := eniIPs[allocation.IPv4]
-		if eni == nil {
-			//Check if this IP belongs to prefix, this can happen when knob is toggled from enable to disable with pods assigned
-			ds.log.Infof("Got IP to recover %s\n", allocation.IPv4)
-			ipv4Prefix := getPrefixFromIPv4Addr(allocation.IPv4)
-			ds.log.Infof("Retrieved prefix %s", ipv4Prefix.String())
-
-			eni := eniPrefixes[ipv4Prefix.String()]
-			if eni == nil {
-				ds.log.Infof("datastore: Sandbox %s uses unknown IPv4 %s - presuming stale/dead", allocation.IPAMKey, allocation.IPv4)
-				continue
-			}
-			eniPrefixDB := ds.eniPool[eni.ID].IPv4Prefixes[ipv4Prefix.String()]
-
-			eniPrefixDB.UsedIPs++
-			eniPrefixDB.FreeIps--
-
-			err := eniPrefixDB.AllocatedIPs.restorePrefixIP(allocation.IPv4)
-			if err != nil {
-				ds.log.Errorf("Failed to ADD PD IP %s on ENI %s", allocation.IPv4, eni.ID)
-				return err
-			}
-
-			err = ds.AddPrefixIPv4AddressToStore(eni.ID, allocation.IPv4)
-			if err != nil {
-				ds.log.Errorf("Failed to ADD PD IP %s on ENI %s", allocation.IPv4, eni.ID)
-				return err
+		ipv4Addr := net.ParseIP(allocation.IPv4)
+		found := false
+	eniloop:
```

Comment:
it feels to me that we can make this more maintainable by abstract this into a function.

Reply:
Will address this with the next PR for warm ip target support.